src/modals/new-identity-flow/import-identity-steps/manual-recovery/ImportManualRecovery.js

```diff
@@ -0,0 +1,96 @@
+import { FORM_ERROR } from 'final-form';
+import styles from './ImportManualRecovery.css';
+
+const DID_METHOD = 'ipid';
+
+class ImportManualRecovery extends Component {
+    acceptedMnemonic = undefined;
+
+    state = {
+        promise: undefined,
+    };
+
+    render() {
+        const { promise } = this.state;
+
+        return (
+            <div className={ styles.contentWrapper }>
+                <h2 className={ styles.heading }>Enter Secret Recovery Key</h2>
+                <p>
+                    Please enter, in the correct order, the 12 words you were asked
+                    to save as your paper key.
+                </p>
+                <Form onSubmit={ this.handleSubmit }>
+                    {({ handleSubmit, hasValidationErrors, submitError }) => (
+                        <form autoComplete="off" onSubmit={ handleSubmit } className={ styles.form }>
+                            <RecoveryTextArea className={ styles.input } />
+                            <div className={ styles.advance }>
+                                { submitError &&
+                                    <FeedbackMessage className={ styles.error } type="error" variant="large">
+                                        Oops, seems that recovery key is not valid, please try again.
+                                    </FeedbackMessage>
+                                }
+                                <ButtonPromiseState
+                                    promise={ promise }
+                                    onSettle={ this.handleSettle }>
+                                    { ({ status }) => (
+                                        <Button
+                                            disabled={ hasValidationErrors }
+                                            feedback={ status }>
+                                                Restore
+                                        </Button>
+                                    )}
+                                </ButtonPromiseState>
+                            </div>
+                        </form>
+                    )}
+                </Form>
+            </div>
+        );
+    }
+
+    handleSubmit = (data) => {
+        const { mnemonic } = data;
+
+        if (getPromiseState(this.state.promise).status === 'pending') {
+            return;
+        }
+
+        this.acceptedMnemonic = mnemonic;
+        const promise = this.props.peekIdentity(DID_METHOD, { mnemonic });
+
+        this.setState({ promise });
+
+        return promise.catch((err) => {
+            console.error(err);
+
+            return {
+                [FORM_ERROR]: err.message,
+            };
+        });
+    };
+
+    handleSettle = (state) => {
+        const { nextStepId } = this.props;
+
+        if (state.status === 'fulfilled') {
+            this.props.onNextStep(nextStepId, { identityDetails: state.value.details, mnemonic: this.acceptedMnemonic });
```

Comment:
Change `state.value.details` to `state.value.profileDetails`. Perhaps `identityDetails` should be renamed accordingly too.